tests/ops/einsum/test_adjoint.py

```diff
@@ -114,3 +115,24 @@ def test_marginal(equation):
         )
         actual = operand._pyro_backward_result
         assert_equal(expected, actual)
+
+
+def test_require_backward_memory_leak():
+    # Using clone resolves memory leak.
+    for i in range(10):
+        x = torch.tensor(0.0)
+        require_backward(x)
+        x._pyro_backward.process(x.clone())
+
+    tensors = [o for o in gc.get_objects() if torch.is_tensor(o)]
+    assert len(tensors) <= 5
```

Comment:
Could we grab the initial tensor count and use relative assertions in this test? Im afraid the current absolute assertions might fail if someome adds new global tensors, e.g. in a `@pytest.mark.parametrize` list or in a global cache

Reply:
Maybe this is why ci is failing

Reply:
That's a good point! I just updated the test to incorporate the logic. CI seems to be failing due to a deprecation warning caused by `torch.is_tensor(o)` during gc.